hazelcast/src/main/java/com/hazelcast/jet/impl/JetClientInstanceImpl.java

```diff
@@ -140,21 +145,64 @@ public Job newJobProxy(long jobId, boolean isLightJob, @Nonnull Object jobDefini
         return new ClientJobProxy(client, jobId, isLightJob, jobDefinition, config);
     }
 
-    @Override
+    /**
+     * For the client side, the jar is uploaded to a random cluster member and then this member runs the main method to
+     * start the job.
+     * The jar should have a main method that submits a Pipeline with {@link #newJob(Pipeline)} or
+     * {@link #newLightJob(Pipeline)} methods
+     * <p>
+     * The upload operation is performed in parts to avoid OOM exceptions on the client and member.
+     * For Java clients the part size is controlled by {@link com.hazelcast.client.properties.ClientProperty#JOB_UPLOAD_PART_SIZE}
+     * property
+     * <p>
+     * Limitations for the client side jobs:
+     * <ul>
+     *     <li>The job can only access resources on the member or cluster. This is different from the jobs submitted from
+     *     the hz-cli tool. A job submitted from hz-cli tool creates a local HazelcastInstance on the client JVM and
+     *     connects to cluster. Therefore, the job can access local resources. This is not the case for the jar
+     *     uploaded to a member.
+     *     </li>
+     * </ul>
+     *
+     * @throws JetException on error
+     */
     public void submitJobFromJar(@Nonnull SubmitJobParameters submitJobParameters) {
```

Comment:
So, is this method part of the private API now?

Reply:
Yes it is private API now.  Moved the SubmitJobParameters and tests to private impl package. 
Rev : b6b13939553a79ec6b41f39fd0b3a2da2b03d5fe